File: lab3/task_12346/cs/Circle.cs
using System;

namespace Lab3 {

public class Circle
{
    private double radius;
    private double x;
    private double y;

    public double Radius {
      get { return radius; }
      set { radius = value; }
    }

    public double X {
      get { return x; }
      set { x = value; }
    }

    public double Y {
      get { return y; }
      set { y = value; }
    }

    public Circle()
    {
        radius = 0;
        x = 0;
        y = 0;
    }

    public Circle(double r)
    {
        radius = r;
        x = 0;
        y = 0;
    }

    public Circle(double r, double arg_x, double arg_y)
    {
        radius = r;
        x = arg_x;
        y = arg_y;
    }

    public void Init(double r, double arg_x, double arg_y)
    {
        Radius = r;
        x = arg_x;
        y = arg_y;
    }

    public void Read()
    {
        Console.Write("Введите радиус: ");
        radius = Convert.ToDouble(Console.ReadLine());
        Console.Write("Введите координату x: ");
        x = Convert.ToDouble(Console.ReadLine());
        Console.Write("Введите координату y: ");
        y = Convert.ToDouble(Console.ReadLine());
    }

    public void Display()
    {
        Console.WriteLine($"Радиус: {radius}");
        Console.WriteLine($"Координата x: {x}");
        Console.WriteLine($"Координата y: {y}");
    }

    public double Distance()
    {
      try {
        if (radius < 0) {
          throw new Exception("Радиус не может быть отрицательным");
        }
        if (radius == 0) {
          throw new Exception("Радиус не может быть равен нулю");
        }
      } catch (Exception e) {
        Console.WriteLine(e.Message);
        return 0;
      }
        return Math.Sqrt(x * x + y * y);
    }

    public static double StaticDistance(Circle c)
    {
        return Math.Sqrt(c.x * c.x + c.y * c.y);
    }

    public void CircleRef(ref double distance)
    {
        distance = Math.Sqrt(x * x + y * y);
    }

    public void CircleOut(out double distance)
    {
        distance = Math.Sqrt(x * x + y * y);
    }

    public static Circle Add(Circle c1, Circle c2)
    {
        Circle result = new Circle();
        result.Init(c1.radius + c2.radius, c1.x + c2.x, c1.y + c2.y);
        return result;
    }

    public Circle add(Circle c)
    {
        Circle result = new Circle();
        result.Init(this.radius + c.radius, this.x + c.x, this.y + c.y);
        return result;
    }
}
}

File: lab3/task_12346/cpp/Circle.cpp
#include <iostream>
using namespace std;

class Circle {
  private:
    double radius;
    double x;
    double y;

  public:
    Circle();
    Circle(double r);
    Circle(double r, double x, double y);

    void Init(double r, double x, double y);
    void Read();
    void Display();

    double distance();
    void distance(double& d);
    void distance(double* d);
    static double staticDistance(Circle c);

    static Circle staticAdd(Circle c1, Circle c2);
    Circle add(Circle c2);
};

  Circle::Circle() {
    this->Init(0, 0, 0);
  }

  Circle::Circle(double r) {
    this->Init(r, 0, 0);
  }

  Circle::Circle(double r, double x, double y) {
    this->Init(r, x, y);
  }

  void Circle::Init(double r, double arg_x, double arg_y) {
    radius = r;
    x = arg_x;
    y = arg_y;
  }

  void Circle::Read() {
    cout << "Введите радиус:";
    cin >> radius;
    cout << "Введите координату x:";
    cin >> x;
    cout << "Введите координату y:";
    cin >> y;
  }

  void Circle::Display() {
    cout << "Радиус: " << radius << endl;
    cout << "Координата x: " << x << endl;
    cout << "Координата y: " << y << endl;
  }

  double Circle::distance() {
    try {
      if (radius < 0) {
        throw "Радиус не может быть отрицательным";
      }
      if (radius == 0) {
        throw "Радиус не может быть равен нулю";
      }
    } catch (const char* msg) {
      cout << msg << endl;
      return -1;
    }

    return sqrt(x * x + y * y);
  }

  void Circle::distance(double& d) {
    d = distance();
  }

  void Circle::distance(double* d) {
    *d = distance();
  }

  double Circle::staticDistance(Circle c) {
    return c.distance();
  }

  Circle Circle::staticAdd(Circle c1, Circle c2) {
    Circle result;
    result.Init(c1.radius + c2.radius, c1.x + c2.x, c1.y + c2.y);
    return result;
  }

  Circle Circle::add(Circle c2) {
    this->radius = this->radius + c2.radius;
    this->x = this->x + c2.x;
    this->y = this->y + c2.y;

    Circle result;
    result.Init(this->radius, this->x, this->y);
    return result;
  }